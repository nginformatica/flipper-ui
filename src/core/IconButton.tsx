import MuiIconButton from '@material-ui/core/IconButton'
import React, { MouseEvent } from 'react'
import { DefaultProps } from './types'

export interface IconButtonProps extends DefaultProps {
    color?: 'default' | 'inherit' | 'primary' | 'secondary'
    disabled?: boolean
    onClick?(event: MouseEvent<HTMLButtonElement>): void
}

const IconButton = ({
    children,
    padding,
    margin,
    style,
    ...otherProps
}: IconButtonProps) => (
    <MuiIconButton {...otherProps} style={{ margin, padding, ...style }}>
        {children}
    </MuiIconButton>
)

export default IconButton
