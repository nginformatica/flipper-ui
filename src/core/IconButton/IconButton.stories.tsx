import { Add as IconAdd } from '@/icons'
import { Meta, StoryFn } from '@storybook/react'
import React from 'react'
import IconButton from '.'

export default {
    title: 'IconButton',
    component: IconButton
} as Meta<typeof IconButton>

const Template: StoryFn<typeof IconButton> = args => <IconButton {...args} />

export const Default = Template.bind({})
Default.args = {
    children: <IconAdd />
}

export const Primary = Template.bind({})
Primary.args = {
    color: 'primary',
    children: <IconAdd />
}

export const Secondary = Template.bind({})
Secondary.args = {
    color: 'secondary',
    children: <IconAdd />
}

export const Disabled = Template.bind({})
Disabled.args = {
    disabled: true,
    children: <IconAdd />
}
