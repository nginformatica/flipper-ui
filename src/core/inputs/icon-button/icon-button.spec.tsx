import * as React from 'react'
import { render, screen } from '@testing-library/react'
import IconButton from '.'

describe('IconButton', () => {
    it('should render', () => {
        render(<IconButton>IconButton</IconButton>)

        expect(screen.getByText('IconButton')).toBeDefined()
    })

    it('should render with custom style', () => {
        render(
            <IconButton
                data-testid='icon-button-container'
                margin={10}
                padding={5}
                style={{ backgroundColor: 'blue' }}>
                <>IconButton</>
            </IconButton>
        )

        const container = screen.getByTestId('icon-button-container')

        expect(container).toHaveProperty('style.margin', '10px')
        expect(container).toHaveProperty('style.padding', '5px')
        expect(container).toHaveProperty('style.backgroundColor', 'blue')
    })

    it('should match snapshot', () => {
        const { container } = render(<IconButton>IconButton</IconButton>)

        expect(container).toMatchSnapshot()
    })
})
