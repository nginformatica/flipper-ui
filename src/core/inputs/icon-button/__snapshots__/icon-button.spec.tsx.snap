// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    IconButton
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
